tests/test_cognitoidp/test_cognitoidp.py

```diff
@@ -1603,6 +1603,27 @@ def test_sign_up():
     result["UserSub"].should_not.be.none
 
 
+@mock_cognitoidp
+def test_sign_up_existing_user():
+    conn = boto3.client("cognito-idp", "us-west-2")
+    user_pool_id = conn.create_user_pool(PoolName=str(uuid.uuid4()))["UserPool"]["Id"]
+    client_id = conn.create_user_pool_client(
+        UserPoolId=user_pool_id, ClientName=str(uuid.uuid4()),
+    )["UserPoolClient"]["ClientId"]
+    username = str(uuid.uuid4())
+    password = str(uuid.uuid4())
+
+    conn.sign_up(ClientId=client_id, Username=username, Password=password)
+
+    caught = False
+    try:
```

Comment:
Can we use the pytest.raises construct here, instead of a try-except? That's how most of our negative tests are handled.
```
 with pytest.raises(ClientError) as ex:
        ...
err = ex.value.response["Error"]
err["Code"].should.equal("...")
err["Message"].should.equal(...)
```